src/server/src/main/java/io/cassandrareaper/service/RepairManager.java

```diff
@@ -82,11 +85,28 @@ public void resumeRunningRepairRuns(AppContext context) throws ReaperException {
     heartbeat(context);
     Collection<RepairRun> running = context.storage.getRepairRunsWithState(RepairRun.RunState.RUNNING);
     for (RepairRun repairRun : running) {
-      if (!repairRunners.containsKey(repairRun.getId())) {
-        Collection<RepairSegment> runningSegments
-            = context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
+      Collection<RepairSegment> runningSegments =
+          context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
+
+      if (context.storage instanceof IDistributedStorage
+          || !repairRunners.containsKey(repairRun.getId())) {
+        // When multiple Reapers are in use, we can get stuck segments when one instance is rebooted
+        // Any segment in RUNNING state but with no leader should be killed
+        List<UUID> activeLeaders =
+            context.storage instanceof IDistributedStorage
+                ? ((IDistributedStorage) context.storage).getLeaders()
+                : Collections.emptyList();
+
+        abortSegments(
+            runningSegments
+                .stream()
+                .filter(segment -> !activeLeaders.contains(segment.getId()))
+                .collect(Collectors.toSet()),
+            context,
+            repairRun);
+      }
```

Comment:
trivial: could we put this chunk into a private method with a descriptive name? could make this already long method a bit more readable (without resorting to using comments)

Reply:
Done and force pushed